server/server.js

```diff
@@ -48,24 +80,51 @@ app.get('/reviews/all/:productId', (req, res) => {
             const { id, user_picture, review_picture } = photos;
             photosById[id] = { user_picture, review_picture };
           });
-
+          const itemNameById = {};
+          itemDetailsResponse.data.forEach((product) => {
```

Comment:
It's not clear to me why you would need to get the itemName for more than one product at a time. Or maybe I just am not following what's going on here.

Reply:
Because it could be the store where you need it for all - to optimize my 200 lines of backend code I made one component and one handler since they look the same just rendering different data.